crates/core/src/config/secret.rs

```diff
@@ -0,0 +1,259 @@
+use std::path::Path;
+
+use aes_gcm::KeyInit;
+use blake3::traits::digest::generic_array::GenericArray;
+use chacha20poly1305::{XChaCha20Poly1305, XNonce};
+use freenet_stdlib::client_api::DelegateRequest;
+
+use super::*;
+
+const NONCE_SIZE: usize = 24;
+const CIPHER_SIZE: usize = 32;
+
+const NONCE_FILENAME: &str = "nonce";
+const CIPHER_FILENAME: &str = "cipher";
+const TRANSPORT_KEYPAIR_FILENAME: &str = "keypair";
+
+impl ConfigArgs {
+    pub(super) fn read_secrets(
+        path_to_key: PathBuf,
+        path_to_nonce: PathBuf,
+        path_to_cipher: PathBuf,
+    ) -> std::io::Result<Secrets> {
+        let transport_keypair = read_transport_keypair(&path_to_key)?;
+        let nonce = read_nonce(&path_to_nonce)?;
+        let cipher = read_cipher(&path_to_cipher)?;
+
+        Ok(Secrets {
+            transport_keypair,
+            transport_keypair_path: path_to_key,
+            nonce,
+            nonce_path: path_to_nonce,
+            cipher,
+            cipher_path: path_to_cipher,
+        })
+    }
+}
+
+#[derive(Debug, Default, Clone, clap::Parser, serde::Serialize, serde::Deserialize)]
+pub struct SecretArgs {
+    /// Path to the transport keypair file.
+    #[clap(value_parser, env = "TRANSPORT_KEYPAIR")]
+    pub transport_keypair: Option<PathBuf>,
+
+    /// Path to the nonce file.
+    #[clap(value_parser, env = "NONCE")]
+    pub nonce: Option<PathBuf>,
+
+    /// Path to the cipher file.
+    #[clap(value_parser, env = "CIPHER")]
+    pub cipher: Option<PathBuf>,
+}
+
+impl SecretArgs {
+    pub(super) fn build(self, secrets_dir: impl AsRef<Path>) -> std::io::Result<Secrets> {
+        let secrets_dir = secrets_dir.as_ref();
+        let transport_key = self
+            .transport_keypair
+            .as_ref()
+            .map(read_transport_keypair)
+            .transpose()?;
+        let (transport_keypair_path, transport_keypair) = if let Some(transport_key) = transport_key
+        {
+            (self.transport_keypair.unwrap(), transport_key)
+        } else {
+            // try secret dir
+            let path = secrets_dir.join(TRANSPORT_KEYPAIR_FILENAME);
+            if path.exists() {
+                let kp = read_transport_keypair(&path)?;
+                (path, kp)
+            } else {
+                let transport_key = TransportKeypair::new();
+                (path, transport_key)
+            }
+        };
+        let nonce = self.nonce.as_ref().map(read_nonce).transpose()?;
+        let (nonce_path, nonce) = if let Some(nonce) = nonce {
+            (self.nonce.unwrap(), nonce)
+        } else {
+            // try secret dir
+            let path = secrets_dir.join(NONCE_FILENAME);
+            if path.exists() {
+                let nonce = read_nonce(&path)?;
+                (path, nonce)
+            } else {
+                (path, DelegateRequest::DEFAULT_NONCE)
```

Comment:
I suppose this requires updating the stdlib, or is already present there?

Reply:
It is already there.